src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -330,6 +330,78 @@ namespace OpenMS
       return integer_data_arrays_;
     }
 
+    /// Returns an iterator to the first float data array with the given name. 
+    /// The End iterator is returned in case no data array with given name exists.
+    typename FloatDataArrays::iterator getFloatDataArrayByName(const String& name)
```

Comment:
see below, on second thought I think it is the best solution. Still a bit clumsy as we dont really want to iterate anything here. The other solution would be to return an index and return `-1` in case of failure...that also seems pretty straight forward to me...